app/src/main/java/com/lastaoutdoor/lasta/ui/MainActivity.kt

```diff
@@ -1,18 +1,69 @@
 package com.lastaoutdoor.lasta.ui
 
+import android.app.LocaleManager
+import android.os.Build
 import android.os.Bundle
+import android.os.LocaleList
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import androidx.annotation.RequiresApi
+import androidx.core.os.LocaleListCompat
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.lastaoutdoor.lasta.data.db.UserDBRepositoryImpl
+import com.lastaoutdoor.lasta.data.preferences.PreferencesRepositoryImpl
+import com.lastaoutdoor.lasta.di.NetworkModule.provideFirebaseFirestore
 import com.lastaoutdoor.lasta.ui.navigation.AppNavGraph
 import com.lastaoutdoor.lasta.ui.theme.LastaTheme
+import com.lastaoutdoor.lasta.viewmodel.PreferencesViewModel
 import dagger.hilt.android.AndroidEntryPoint
+import java.util.Locale
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+  @RequiresApi(Build.VERSION_CODES.TIRAMISU)
   override fun onCreate(savedInstanceState: Bundle?) {
     enableEdgeToEdge()
     super.onCreate(savedInstanceState)
+
```

Comment:
If you are able to comment a bit more this part of the code it would be great :)